satpy/writers/__init__.py

```diff
@@ -865,14 +869,17 @@ def separate_init_kwargs(cls, kwargs):
         return init_kwargs, kwargs
 
     def save_dataset(self, dataset, filename=None, fill_value=None,
-                     overlay=None, decorate=None, compute=True, **kwargs):
+                     overlay=None, decorate=None, compute=True, units=None, **kwargs):
         """Save the ``dataset`` to a given ``filename``.
 
         This method creates an enhanced image using :func:`get_enhanced_image`.
         The image is then passed to :meth:`save_image`. See both of these
         functions for more details on the arguments passed to this method.
 
         """
+        if units is not None:
+            import pint_xarray  # noqa
+            dataset = dataset.pint.quantify().pint.to(units)
```

Comment:
```suggestion
            dataset = dataset.pint.quantify().pint.to(units).pint.dequantify()
```

this way the `units` attribute is retained.  I'm not sure if it matters for GeoTIFF, but it does for the `ninjogeotiff` writer, which inherits from the `GeoTIFF` writer, and of course it matters for the CF writer.

Reply:
Warning: unit conversion changes any integer data type to float64.  This is due to pint, not Satpy, but in Satpy this happens immediately before storing.  I wonder if there are cases where this could lead to unexpected consequences.

Reply:
Good point, I will think about this!